### PR TITLE
(WIP) tRPC router methods for users

### LINKED ISSUE #11: tRPC router for users
Create a tRPC router for users CRUD, creating a mentor and a mentee. This involves creating a profile from inputs to the router functions. These don't have to be the end-all-be-all for the endpoints, but will provide a good starting point for hooking up with forms and tables.

https://trpc.io/docs/server/routers
https://www.prisma.io/docs/concepts/components/prisma-client

### PR DESCRIPTION
**Pull request by nad2040**
Closes #11

Added:
- getById
- getAll

### CODE REVIEW

File: src/server/trpc.ts
Comment (ulrokx) [inline src/server/trpc.ts:1]: Can you move this to `src/server/api/routers`? The example router in that folder is how you should create this user router.

File: src/server/trpc.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const t = initTRPC.context<User>().create();
+
+export const router = t.router;
+export const publicProcedure = t.procedure;
+
+const appRouter = router({
+  getById: publicProcedure.query(async (opts) => {
+    const userId = opts.ctx.id;
+    return db.user.findUnique({
```
Comment (ulrokx) [inline src/server/trpc.ts:14]: What happens when the user is not found? Can you show what happens when a user is not found and in general how to handle errors like that? tRPC docs will probably be helpful for this

**Review by ulrokx — CHANGES_REQUESTED** [review]

**Comment by nad2040** [comment]
- `npm run dev`
- go to [http://localhost:3000/api/trpc/dbquery.getAll](http://localhost:3000/api/trpc/dbquery.getAll)

File: src/server/api/routers/dbquery.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import {
+  createTRPCRouter,
+  protectedProcedure,
+  publicProcedure,
+} from "~/server/api/trpc";
+
+import { db } from "~/server/db";
+
+export const dbQueryRouter = createTRPCRouter({
```
Comment (ulrokx) [inline src/server/api/routers/dbquery.ts:12]: Can we name this `userRouter` and name the file `user.ts`? We are going to add a lot more routers for different parts of the application, so naming this dbQueryRouter doesn't make much sense. Name it on functionality not implementation.

File: src/server/api/root.ts [hunk 3]
```diff
@@ -8,6 +9,7 @@ import { createTRPCRouter } from "~/server/api/trpc";
  */
 export const appRouter = createTRPCRouter({
   example: exampleRouter,
+  dbquery: dbQueryRouter,
```
Comment (ulrokx) [inline src/server/api/root.ts:12]: This can be `user: userRouter`

**Review by ulrokx — CHANGES_REQUESTED** [review]

File: src/server/api/routers/dbquery.ts (hunk 2 above)
Comment (nad2040) [reply src/server/api/routers/dbquery.ts:12]: Any comments on using protected procedure rather than public procedure?

**Comment by nad2040** [comment]
Any comments on using protected procedure rather than public procedure?

**Comment by ulrokx** [comment]
@nad2040 looks good for now, we can figure out what restricting routes to specific roles look like next.

**Review by ulrokx — APPROVED** [review]

**Pull request merged**